### PR TITLE
read etcd endpoint information

### PR DESCRIPTION
**Pull request by deads2k**
Reads etcd endpoint information for storage.

/assign @mfojtik

### CODE REVIEW

File: pkg/operator/observe_config.go [hunk 1]
```diff
@@ -5,9 +5,12 @@ import (
 	"reflect"
 	"time"
 
+	"k8s.io/apimachinery/pkg/util/diff"
```
Comment (sttts) [inline pkg/operator/observe_config.go]: nit: order

File: pkg/operator/observe_config.go [hunk 2]
```diff
@@ -73,9 +78,29 @@ func (c ConfigObserver) sync() error {
...
+	// Setting them observed config causes the normal reconciliation loop to run
+	etcdURLs := []string{}
+	etcdEndpoints, err := c.kubeClient.CoreV1().Endpoints(etcdNamespaceName).Get("etcd", metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+	if etcdEndpoints != nil {
+		for _, subset := range etcdEndpoints.Subsets {
+			for _, address := range subset.Addresses {
+				etcdURLs = append(etcdURLs, "https://"+address.Hostname+"."+etcdEndpoints.Annotations["alpha.installer.openshift.io/dns-suffix"]+":2379")
```
Comment (sttts) [inline pkg/operator/observe_config.go]: without the annotation we get a trailing dot. Intended?

Comment (deads2k) [reply pkg/operator/observe_config.go]: > without the annotation we get a trailing dot. Intended?

Without the annotation I think we fail in general.

**Comment by deads2k** [comment]
comments addressed

**Comment by deads2k** [comment]
peer already lgtmed.  Tagging this one.

**Pull request merged**